tools/src/main/java/org/neo4j/tools/txlog/CheckTxLogs.java

```diff
@@ -94,20 +109,59 @@ public static void main( String[] args ) throws Exception
     // used in other projects do not remove!
     public boolean checkAll( File storeDirectory ) throws IOException
     {
-        return scan( storeDirectory, new PrintingInconsistenciesHandler( out ), CheckTypes.CHECK_TYPES );
+        PhysicalLogFiles logFiles = new PhysicalLogFiles( storeDirectory, fs );
+        InconsistenciesHandler handler = new PrintingInconsistenciesHandler( out );
+        boolean success = scan( logFiles, handler, CheckTypes.CHECK_TYPES );
+        success &= validateCheckPoints( logFiles, handler );
+        return success;
+    }
+
+    boolean validateCheckPoints( PhysicalLogFiles logFiles, InconsistenciesHandler handler ) throws IOException
+    {
+        long lowestLogVersion = logFiles.getLowestLogVersion();
+        Map<Long,Long> logFileSizes = new HashMap<>();
+        for ( long i = lowestLogVersion; i <= logFiles.getHighestLogVersion(); i++ )
+        {
+            logFileSizes.put( i, fs.getFileSize( logFiles.getLogFileForVersion( i ) ) );
+        }
+
+        boolean success = true;
+        LogEntryCursor logEntryCursor = LogTestUtils.openLogs( fs, logFiles );
+        while ( logEntryCursor.next() )
+        {
+            LogEntry logEntry = logEntryCursor.get();
+            if ( logEntry instanceof CheckPoint )
+            {
+                LogPosition logPosition = logEntry.<CheckPoint>as().getLogPosition();
+                // if the file has been pruned we cannot validate the check point
+                if ( logPosition.getLogVersion() >= lowestLogVersion )
+                {
+                    Long size = logFileSizes.get( logPosition.getLogVersion() );
+                    if ( logPosition.getByteOffset() < 0 || size == null || logPosition.getByteOffset() > size )
+                    {
+                        long currentLogVersion = logEntryCursor.getCurrentLogVersion();
+                        handler.reportInconsistentCheckPoint( currentLogVersion, logPosition, size );
```

Comment:
At this level yes. The `PrintingInconsistenciesHandler` will print a more specific description for the inconsistency at hand.